### PR TITLE
[dict] Refactor the dictionary initialization. NFC.

### PR DESCRIPTION
**Pull request by vgvassilev**
This patch makes the implementation reusable and ready to configure it such
that we can remove the redundant information for modules.

### CODE REVIEW

**Comment by vgvassilev** [comment]
@bellenot, I suspect the windows failure is due to the max_size limit of std::string. Could you confirm?

**Comment by vgvassilev** [comment]
@phsft-bot build!

File: core/dictgen/src/TModuleGenerator.cxx [hunk 1]
```diff
@@ -445,45 +494,29 @@ void TModuleGenerator::WriteRegistrationSource(std::ostream &out,
...
-   out << "namespace {\n"
-       "  void TriggerDictionaryInitialization_"
-       << GetDictionaryName() << "_Impl() {\n"
-       "    static const char* headers[] = {\n";
-   if (fInlineInputHeaders) {
-      out << 0 ;
-   } else {
-      WriteHeaderArray(out);
-   };
+   std::string headerArray = "0";
```
Comment (Axel-Naumann) [inline core/dictgen/src/TModuleGenerator.cxx]: Should this be `= "nullptr"`?

Comment (vgvassilev) [reply core/dictgen/src/TModuleGenerator.cxx]: Yes, it should, however, the makepch.py has a regex based on this and I failed to adapt it...

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
@phsft-bot build just on windows10/default

**Comment by vgvassilev** [comment]
@phsft-bot build just on windows10/default

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Pull request merged**